### ISSUE TITLE
'[tslint] missing whitespace [whitspace]' when setting "nospace" for both sides of colon for "parameter" in "typedef-whitespace"

### ISSUE THREAD

**Issue by lidin** [palantir/tslint#4342]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.2.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
function test(arg:string) {
    console.log(arg);
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "typedef-whitespace": {
            "options": [{
                "parameter": "nospace"
            }, {
                "parameter": "nospace"
            }]
        },
        "whitespace": {
            "options": [
                "check-type"
            ]
        }
    }
}
```

Here I am using "whitespace" to get warnings about wrongly formatted type declarations. Omitting the "whitespace" options disables type format checking, which is not the behavior I want. How can I get TSLint to warn me about type formats without yielding this error?

#### Actual behavior
Warning: [tslint] missing whitespace [whitspace].

#### Expected behavior
No warnings.

**Comment by JoshuaKGoldberg** [comment]
@lidin is your issue that you don't want a space after the `:`, and the `whitespace` rule always adds it even though the `typedef-whitespace` has the `nospace` setting?

Seems like this is a bug in how `whitespace` is configured; should it have the ability to have a `"check-type": false`? Is such a thing already possible with the new options format (#889)?

Note that TSLint core is moving away from enforcing whitespace settings (#846, for example). These rules are still supported and bug fixes such as this one can be taken in, but the general recommendation is to switch to something like [Prettier](https://prettier.io/).

**Comment by lidin** [comment]
Yes, I expected it to always warn me about whenever I have any space between the parameter name, colon, and type. With the whitespace setting I get the other warning, and without it, I don’t get any warnings about it at all.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
